agent/dns.go

```diff
@@ -1077,7 +1077,7 @@ func trimUDPResponse(req, resp *dns.Msg, udpAnswerLimit int) (trimmed bool) {
 	// uncompresses them.
 	// Even when size is too big for one single record, try to send it anyway
 	// (useful for 512 bytes messages)
-	for len(resp.Answer) > 1 && resp.Len() > maxSize {
+	for len(resp.Answer) > 1 && resp.Len() > maxSize-7 {
```

Comment:
why 7?

Reply:
https://github.com/miekg/dns/blob/9b7437f11d75979ac166ebd0780edd96e62d5145/dns.go#L64-L70
```
type RR_Header struct {
	Name     string `dns:"cdomain-name"`
	Rrtype   uint16
	Class    uint16
	Ttl      uint32
	Rdlength uint16 // Length of data after header.
}
```
The overflow fails on the TTL read, which is 4 bytes, the unpacking code checks if we have 4 bytes extra to read otherwise overflows and after that we still need 2 bytes for Rrdlength.
So the total is 6 bytes